clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -265,6 +280,8 @@ public MembershipManagerImpl(String groupId,
         this.assignmentReadyToReconcile = new TreeSet<>(TOPIC_ID_PARTITION_COMPARATOR);
         this.currentAssignment = new HashMap<>();
         this.log = logContext.logger(MembershipManagerImpl.class);
+        this.stateUpdatesListeners = new ArrayList<>();
+        this.time = Time.SYSTEM;
```

Comment:
You probably want to unit-test this class with mocktime instead of system time, so maybe pass this from the outside.